test/distributed/test_distributed.py

```diff
@@ -2208,6 +2210,62 @@ def test_SyncBatchNorm_process_group(self):
         process_group_sync = res50_model_sync.layer1[0].bn1.process_group
         self.assertEqual(process_group_sync, process_group)
 
+    def test_DistributedSampler_padding(self):
```

Comment:
nit: I'm wondering whether the test can be moved to data loader related test file?

Reply:
I'd prefer to leave it in this fiile as `test_dataloader` doesn't have the multiprocessing setup, and we have the support for it in `test_distributed`, which we need since the test uses distributed collectives. Let me know if it's better to have this in the dataloader test though, and we can probably add a basic multiprocessing setup in the test to enable that.